projects/open-training/packages/Training/src/Entity/Trainer.php

```diff
@@ -5,12 +5,17 @@
 use Doctrine\Common\Collections\ArrayCollection;
 use Doctrine\Common\Collections\Collection;
 use Doctrine\ORM\Mapping as ORM;
+use OpenTraining\Entity\UploadableImageTrait;
+use Vich\UploaderBundle\Mapping\Annotation as Vich;
 
 /**
  * @ORM\Entity
+ * @Vich\Uploadable
  */
 class Trainer
 {
+    use UploadableImageTrait;
+
     /**
      * @ORM\Id()
      * @ORM\GeneratedValue()
```

Comment:
@alterphp Hi, I tried to integrate [VichUploaderBundle](https://symfony.com/doc/master/bundles/EasyAdminBundle/integration/vichuploaderbundle.html) to EAB. It was pain to use and configure manually for each grid and forms...

So I've added trait + ConfigPass drop-in support for uploadable images. 

- Trait: https://github.com/TomasVotruba/open-project/pull/29/files#diff-7b21ea5c35990439b5c4b3ed87fa8bee 
- Config Pass: https://github.com/TomasVotruba/open-project/pull/29/files#diff-9eddc9a2014f594b78076a50a73f9c36

All we need to do is active the config:

```yaml
services:

    OpenTraining\EasyAdmin\ConfigPass\ImagePropertyAutoconfigureConfigPass:
        tags:
            - { name: 'easyadmin.config_pass', priority: 2 }
```

And decorate entity with trait and annotation like this line comment.

<br>

How useful this could be in EasyAdminExtension?

Reply:
It's related to single-image use only. For more complex needs, custom code is needed. 

But for single-image, current setup is overkill:

- define property
- define setters
- activate random property override so the image is loaded - this is big WTF that took me 30 minutes to make work
- configure image for grids - all of them! → figure out what are their names
- configure imageFile for form - all of them! → -||-
- use explicit `fields`, because `exclude_fields` cannot be used - so the whole `exclude_fields` idea is killed everywhere with image

Reply:
This was setup here at 1st usage. So much unneccesary complexity

```diff
+parameters:
+    training_grid_fields:
+        - 'name'
+        - { property: 'image', type: 'image', base_path: '%training_image_path%' }
+        - 'perex'
+        - 'description'
+        - 'capacity'
+        - 'price'
+        - 'place'
+        - 'trainer'

easy_admin:
    entities:
        Training:
            label: 'Školení'
            class: 'OpenTraining\Training\Entity\Training'
            show: 
-               exclude_fields:
-                   - terms
+               fields: '%training_grid_fields%'
            list: 
-               exclude_fields:
-                   - terms
+               fields: '%training_grid_fields%'
```

Reply:
Yes but ... doing this way is like moving complexity to an extension bundle, by pushing it a new feature that increases its dependency map (EasyAdminExtensionBundle would now depend on VichUploaderBundle too).

To me, `uploadedAt` should be a metadata directly handled (as a \DateTimeImmutable) by VichUploaderBundle itself, as well as `size`, `mimeType` and `dimensions`. So that you don't need need a third property, second persisted one, to deal with an image...